server/insights/src/test/java/org/infinispan/server/insights/InsightsActivationTest.java

```diff
@@ -1,62 +1,113 @@
 import org.infinispan.factories.GlobalComponentRegistry;
 import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.persistence.dummy.DummyInMemoryStoreConfigurationBuilder;
 import org.infinispan.server.insights.config.InsightsActivation;
 import org.infinispan.test.AbstractInfinispanTest;
 import org.infinispan.test.fwk.TestCacheManagerFactory;
 import org.testng.ITestContext;
 import org.testng.annotations.AfterClass;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
-import com.redhat.insights.config.EnvAndSysPropsInsightsConfiguration;
 import com.redhat.insights.reports.InsightsReport;
 
-@Test(groups = "functional", testName = "server.insights.InsightsActivationTest")
+@Test(groups = "functional", testName = "server.insights.InsightsActivationTest", dataProvider = "optOut")
 public class InsightsActivationTest extends AbstractInfinispanTest {
 
-   public void enabled() throws Exception {
+   private static final String RHEL_MACHINE_ID_FILE_PATH
+         = InfinispanInsightsConfiguration.ENV_RHEL_MACHINE_ID_FILE_PATH.toLowerCase().replace('_', '.');
+   private static final String ARCHIVE_UPLOAD_DIR
+         = InfinispanInsightsConfiguration.ENV_ARCHIVE_UPLOAD_DIR.toLowerCase().replace('_', '.');
+   private static final String OPT_OUT
+         = InfinispanInsightsConfiguration.ENV_OPT_OUT.toLowerCase().replace('_', '.');
+
+   public void enabled(boolean optOut) throws Exception {
       Path tmpDir = Paths.get(System.getProperty("java.io.tmpdir"), "blablabla");
       Files.createDirectories(tmpDir);
       System.setProperty(InsightsModule.REDHAT_INSIGHTS_ACTIVATION_PROPERTY_NAME, InsightsActivation.ENABLED.name());
-      System.setProperty(InfinispanInsightsConfiguration.ENV_RHEL_MACHINE_ID_FILE_PATH.toLowerCase().replace('_', '.'), tmpDir.toString());
-      System.setProperty(EnvAndSysPropsInsightsConfiguration.ENV_ARCHIVE_UPLOAD_DIR.toLowerCase().replace('_', '.'), tmpDir.toString());
+      System.setProperty(RHEL_MACHINE_ID_FILE_PATH, tmpDir.toString());
+      System.setProperty(ARCHIVE_UPLOAD_DIR, tmpDir.toString());
+      System.setProperty(OPT_OUT, optOut + "");
       try (EmbeddedCacheManager cacheManager = TestCacheManagerFactory.createServerModeCacheManager()) {
          InsightsService service = GlobalComponentRegistry.of(cacheManager).getComponent(InsightsService.class);
          assertThat(service).isNotNull();
-         InsightsReport report = service.report();
-         String json = report.serialize();
-         assertThat(json).isNotEmpty();
+         try (InsightsReport report = service.report()) {
+            assertThat(report).isNotNull();
+         }
+         // Don't serialize the report here to avoid conflicts with the concurrent serialization made by the Insights client
+
+         if (!optOut) {
+            // with this option the Insights Java client (not our code) is not allowed to produce the report even locally
```

Comment:
This comment seems to be contradicting the if condition and the block

Reply:
in this case I'm referring to the Insights client report production to the filesystem

Reply:
Anyway, I updated the test comment.

Reply:
Thanks, I found confusing that the comment was about `optOut` set to true while being within the block that executed only when `optOut` was set to false